Corona-Warn-App/src/androidTest/java/testhelpers/viewmodels/MockViewModelModule.kt

```diff
@@ -1,35 +1,16 @@
 package testhelpers.viewmodels
 
-import android.os.Bundle
-import androidx.lifecycle.SavedStateHandle
-import androidx.savedstate.SavedStateRegistryOwner
 import dagger.Module
 import dagger.Provides
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
 import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactory
-import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
 
 @Module
 class MockViewModelModule {
 
     @Provides
-    fun viewmodelFactoryProvider(): CWAViewModelFactoryProvider.Factory {
-        val factory = object : CWAViewModelFactoryProvider.Factory {
-            override fun create(
-                savedStateOwner: SavedStateRegistryOwner,
-                defaultSavedState: Bundle?,
-                assistAction: ((CWAViewModelFactory<out CWAViewModel>, SavedStateHandle) -> CWAViewModel)?
-            ): CWAViewModelFactoryProvider {
-                return CWAViewModelFactoryProvider(
-                    CREATORS,
-                    savedStateOwner,
-                    defaultSavedState,
-                    assistAction
-                )
-            }
-        }
-        return factory
-    }
+    fun viewModelCreators(): MutableMap<Class<out CWAViewModel>, CWAViewModelFactory<out CWAViewModel>> =
```

Comment:
What happened here?
How does it relate tor changing the assisted injection library?

Reply:
Dagger was throwing a compile-time error: `Dagger does not support AssistedInject types` pointing to the `CWAViewModelFactoryProvider` constructor.  in UI tests we provide it through a module, while it is not the same in AppComponent. I assumed it does not work with @Provides. so instead of providing an instance of CWAViewModelFactoryProvider. I provided the creators map instead, and it seems to be fine. if you have a better way, I will be happy to do it.

Reply:
Exact error `Dagger does not support providing @AssistedFactory types.`

Reply:
I'm just suprised this works, I prefer your solution, I'm sure I tried it previously though and couldn't get it to work.

So if it all still works, this is nice :+1: